File: lib/go/rpc/RpcHeader.go
package rpc

const RpcHeader_strFingerprint = "2bcc93ddec2a7ac9beece683a78b5561"

type RpcHeader struct {
	Version     int8   `json:"version"`
	MsgType     int16  `json:"msgType"`
	ServiceName string `json:"serviceName"`
}

func (this *RpcHeader) GetFingerprint() string {
	return RpcHeader_strFingerprint
}
func (this *RpcHeader) Serialize(P__ IProtocol) {
	P__.WriteInt8(this.Version)

	P__.WriteInt16(this.MsgType)

	P__.WriteString(this.ServiceName)

}
func (this *RpcHeader) DeSerialize(P__ IProtocol) error {
	this.Version = P__.ReadInt8()

	this.MsgType = P__.ReadInt16()

	this.ServiceName = P__.ReadString()

	return nil
}

File: lib/ts/rpc/RpcHeader.ts
import  * as rpc from "../rpc/index"
import  {MsgProtocolType}  from "./MsgProtocolType"
export class RpcHeader
{ 
	public static readonly strFingerprint:string="2bcc93ddec2a7ac9beece683a78b5561"
	public version:number=0
	public msgType:MsgProtocolType=0
	public serviceName:string=""
	//serialize
	public serialize( __P__:rpc.Protocol):void 
	{ 
		__P__.writeInt8(this.version);

		__P__.writeInt16(this.msgType);

		__P__.writeString(this.serviceName);

	}// serialize
	//deSerialize
	public deSerialize( __P__:rpc.Protocol):void 
	{ 
		this.version=__P__.readInt8()

		this.msgType=__P__.readInt16()

		this.serviceName=__P__.readString()

	}// deSerialize
}//class
